src/main/java/org/traccar/api/resource/PositionResource.java

```diff
@@ -86,6 +88,25 @@ public Collection<Position> getJson(
         }
     }
 
+    @DELETE
+    public Response remove(
+            @QueryParam("deviceId") long deviceId,
+            @QueryParam("from") Date from, @QueryParam("to") Date to) throws StorageException {
+        if (from == null || to == null || deviceId <= 0) {
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
```

Comment:
We usually don't validate input parameters like this. It will fail later or if the parameters are wrong/missing.

Reply:
After reviewing other endpoints I think the following aligns more with the current code standards:
```suggestion
        if (deviceId > 0) {
            permissionsService.checkPermission(Device.class, getUserId(), deviceId);
            if (from != null && to != null) {
                permissionsService.checkRestriction(getUserId(), UserRestrictions::getReadonly);
                ....
            }
        }
```
What do you think?

Reply:
I think we should remove the checks completely.

Reply:
Okay, sounds good.